chainstate/test-suite/src/tests/orders_tests.rs

```diff
@@ -0,0 +1,664 @@
+
+        let tx = TransactionBuilder::new()
+            .add_input(coins_outpoint.into(), InputWitness::NoSignature(None))
+            .add_input(
+                TxInput::AccountCommand(
+                    AccountNonce::new(0),
+                    AccountCommand::FillOrder(order_id, fill_value, Destination::AnyoneCanSpend),
+                ),
+                InputWitness::NoSignature(None),
+            )
+            .add_output(TxOutput::Transfer(
+                OutputValue::TokenV1(token_id, filled_amount),
+                Destination::AnyoneCanSpend,
+            ))
+            .add_output(TxOutput::Transfer(
+                OutputValue::Coin(left_to_fill),
+                Destination::AnyoneCanSpend,
+            ))
+            .build();
+        let partial_fill_tx_id = tx.transaction().get_id();
+        tf.make_block_builder().add_transaction(tx).build_and_process(&mut rng).unwrap();
+
+        assert_eq!(
+            Some(order_data),
+            tf.chainstate.get_order_data(&order_id).unwrap()
+        );
+        assert_eq!(
+            Some(left_to_fill),
+            tf.chainstate.get_order_ask_balance(&order_id).unwrap()
+        );
+        assert_eq!(
+            Some((give_amount - filled_amount).unwrap()),
+            tf.chainstate.get_order_give_balance(&order_id).unwrap()
+        );
+
+        // Fill the rest of the order
+        let fill_value = OutputValue::Coin(left_to_fill);
+        let filled_amount = {
+            let db_tx = tf.storage.transaction_ro().unwrap();
+            let orders_db = OrdersAccountingDB::new(&db_tx);
+            orders_accounting::calculate_fill_order(&orders_db, order_id, &fill_value).unwrap()
+        };
+
+        let tx = TransactionBuilder::new()
+            .add_input(
+                UtxoOutPoint::new(partial_fill_tx_id.into(), 1).into(),
+                InputWitness::NoSignature(None),
+            )
+            .add_input(
+                TxInput::AccountCommand(
+                    AccountNonce::new(1),
+                    AccountCommand::FillOrder(order_id, fill_value, Destination::AnyoneCanSpend),
+                ),
+                InputWitness::NoSignature(None),
+            )
+            .add_output(TxOutput::Transfer(
+                OutputValue::TokenV1(token_id, filled_amount),
+                Destination::AnyoneCanSpend,
+            ))
+            .build();
+        tf.make_block_builder().add_transaction(tx).build_and_process(&mut rng).unwrap();
+
+        assert_eq!(None, tf.chainstate.get_order_data(&order_id).unwrap());
+        assert_eq!(
+            None,
+            tf.chainstate.get_order_ask_balance(&order_id).unwrap()
+        );
+        assert_eq!(
+            None,
+            tf.chainstate.get_order_give_balance(&order_id).unwrap()
+        );
+    });
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn withdraw_order_check_signature(#[case] seed: Seed) {
```

Comment:
We need also tests to cancel (not withdraw, please rename) orders that are partially filled

Reply:
there are tests for that, actual the reason why I chose 'withdraw' was to heighlight that this functionality gets from the order whatever is left

Reply:
We need a test for that here, in this file, at the blockchain level.